models/actions/task_step.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"context"
+	"time"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/timeutil"
+)
+
+// ActionTaskStep represents a step of ActionTask
+type ActionTaskStep struct {
+	ID        int64
+	Name      string `xorm:"VARCHAR(255)"`
+	TaskID    int64  `xorm:"index unique(task_index)"`
+	Index     int64  `xorm:"index unique(task_index)"`
+	RepoID    int64  `xorm:"index"`
+	Status    Status `xorm:"index"`
+	LogIndex  int64
+	LogLength int64
+	Started   timeutil.TimeStamp
+	Stopped   timeutil.TimeStamp
+	Created   timeutil.TimeStamp `xorm:"created"`
+	Updated   timeutil.TimeStamp `xorm:"updated"`
+}
+
+func (step *ActionTaskStep) Duration() time.Duration {
+	return calculateDuration(step.Started, step.Stopped, step.Status)
+}
+
+func init() {
+	db.RegisterModel(new(ActionTaskStep))
+}
+
+func GetTaskStepsByTaskID(ctx context.Context, taskID int64) ([]*ActionTaskStep, error) {
+	var steps []*ActionTaskStep
+	return steps, db.GetEngine(ctx).Where("task_id=?", taskID).OrderBy("number").Find(&steps)
```

Comment:
number has been renamed to index (https://github.com/go-gitea/gitea/commit/d5e6d959ebbf41ad0cb890b8ec8429ea64ae3389)
```suggestion
	return steps, db.GetEngine(ctx).Where("task_id=?", taskID).OrderBy("index").Find(&steps)
```

Reply:
Weird now I get a syntax error index seem to be reserved in sqlite

Reply:
Could you please recreate the db file of sqlite and retry? You know, it's under development, or I must add a migration for changing a field name.

Reply:
> Could you please recreate the db file of sqlite and retry?

My table has the column index, but the sqlite query causes a syntax error, here the same error from the sqlite shell
```
sqlite> select * from action_task_step order by index;
Parse error: near "index": syntax error
  select * from action_task_step order by index;
                            error here ---^
sqlite> select * from action_task_step order by "index";
1|echo 'hello world'
sleep 1
echo 'hello world'
sleep 1
echo 'hello world'
sleep 1
echo 'hello world'
sleep 1
echo 'hello world'
sleep 1
echo 'hello world'
|1|0|1|6|0|0|1672915984|1672914521|1672909714|1672915985
...
```
I need this under sqlite, your db engine might be different and allows unescaped index in the order by clause
```go
return steps, db.GetEngine(ctx).Where("task_id=?", taskID).OrderBy("\"index\"").Find(&steps)
```

_I never found the need for double quotes in the rest of the gitea code, you probably should rename the field again and avoid sqlite keywords_

> You know, it's under development

Yes it is, I regulary recreate my sqlite database. The past days serveral times, because my zombie tasks deadlocked the whole gitea test instance after short time.